app.rb

```diff
@@ -25,120 +34,308 @@ def create_person
-      print 'Name: '
-      name = gets.chomp
-      print 'Specialization: '
-      specialization = gets.chomp
-      { age: age, name: name, specialization: specialization }
+  def create_rental
+    return puts 'There are no books yet.' if @books.empty?
+
+    puts 'Select a book from the following list by number'
+    list_books
+
+    book_choice = gets.chomp.to_i
+
+    if book_choice <= 0 || book_choice > @books.length
+      puts 'Invalid input! Please enter a number within the range.'
+      return
     end
+
+    puts 'Select a person from the following list by number (not id)'
+    list_people
+
+    people_choice = gets.chomp.to_i - 1
+
+    if people_choice.negative? || people_choice >= @people.length
+      puts 'Invalid input! Please enter a number within the range.'
+      return
+    end
+
+    puts 'Date: '
+    date = gets.chomp
+
+    @rentals << Rental.new(date, @books[book_choice - 1], @people[people_choice])
+
+    puts 'Rental created successfully!'
+    save_rentals
   end
 
-  def create_teacher
-    teacher_details = TeacherCreator.fetch_teacher_details
-    person = Teacher.new(teacher_details[:age], teacher_details[:specialization], name: teacher_details[:name])
-    @people.push(person)
-    puts "Teacher '#{teacher_details[:name]}' created successfully"
+  def list_books
+    puts 'List of Books:'
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}. Title: #{book.title}, Author: #{book.author}"
+    end
   end
 
-  class BookCreator
-    def self.fetch_book_details
-      print 'Title: '
-      title = gets.chomp
-      print 'Author: '
-      author = gets.chomp
-      { title: title, author: author }
+  def list_people
+    puts '----- People loaded successfully -----'
+    if File.exist?('people.json')
+      begin
+        people_data = load_people_data
+
+        @people = build_people(people_data).compact
+
+        display_people_info(@people)
+      rescue JSON::ParserError => e
+        puts "Error parsing JSON data: #{e.message}"
+      rescue StandardError => e
+        puts "Error loading data from JSON: #{e.message}"
+      end
+    else
+      puts 'No person data found in people.json'
     end
   end
 
-  def create_book
-    book_details = BookCreator.fetch_book_details
-    book = Book.new(book_details[:title], book_details[:author])
-    @books.push(book)
-    puts "Book '#{book_details[:title]}' created successfully"
+  def load_people_data
+    people_json = File.read('people.json')
+    JSON.parse(people_json)
+  end
+
+  def build_people(people_data)
+    people_data.map.with_index(1) do |person_data, index|
+      build_person_from_data(person_data, index)
+    end
+  end
+
+  def build_person_from_data(person_data, index)
+    case person_data['type']
+    when 'student'
+      build_student(person_data, index)
+    when 'teacher'
+      build_teacher(person_data, index)
+    else
+      puts "Unknown person type: #{person_data['type']}"
+      nil
+    end
```

Comment:
- You have done well so far, but when loading data from the `people.json` file, your app loads duplicate data, this is not the expected behavior of the application. Please fix this 🙏 
![4qa4tq34t](https://github.com/axcell-0/School_library_oop/assets/73607512/7909108a-d791-416f-8187-221f0bd5552b)

**OPTIONAL** - Your app.rb file is quite large, which could make your file less easy to maintain and understand, I suggest you consider refactoring your `app.rb` file. Consider putting the logic for saving and loading data in different files. Please fix this 🙏